### DISCUSSION TITLE
How to enable autosnippets in a kickstart.nvim based init.lua ?

### DISCUSSION THREAD

**Discussion by nchopin** [L3MON4D3/LuaSnip#1244] [Q&A]
Hi, 
Sorry in advance if I'm being dumb. (I read the doc, but to no avail.) My configuration is based on [kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim). I adapted the LuaSnip bit, in order to 
1. add my own snippets, and;
2. enable auto-snippets. 

However, I can't achieve (2); perhaps the issue comes from the way I did (1). 

Here is one of my attempts. Note how I added the `init` entry (in order to add my own snippets), and I tried to enable autosnippets through entry opts. 

```
 "L3MON4D3/LuaSnip",
 opts = { enable_autosnippets = true },    -- added by me
 build = (function()
     -- Build Step is needed for regex support in snippets.
     -- This step is not supported in many windows environments.
     -- Remove the below condition to re-enable on windows.
     if vim.fn.has("win32") == 1 or vim.fn.executable("make") == 0 then
         return
     end
     return "make install_jsregexp"
 end)(),
 init = function()    -- added by me
     require("luasnip.loaders.from_snipmate").lazy_load({ paths = "./snippets/" })   -- added by me
     require("luasnip.loaders.from_lua").lazy_load({ paths = "./lua/luasnippets/" })   -- added by me
 end,    -- added by me
 dependencies = {
     -- `friendly-snippets` contains a variety of premade snippets.
     --    See the README about individual language/framework/plugin snippets:
     --    https://github.com/rafamadriz/friendly-snippets
     {
         "rafamadriz/friendly-snippets",
         config = function()
             require("luasnip.loaders.from_vscode").lazy_load()
         end,
     },
 },
```

I do have access to my snippets, but the "auto-"snippets do not work. I can inspect manually the enable_autosnippets variables, it's set to false. (I can set it to true manually, and then things work.) 

I tried various other things (for instance, using `config` instead of `init` above). I also tried to add the following instruction (as recommended in the doc) at the very end of my `init.lua` file, but still no luck.

`require("luasnip").config.setup({auto_snippets=true})`

* NVIM version: v0.11.0-dev
* luasnip version: stable (automatically updated by Lazy)
* kickstart.nvim: same 

Many thanks in advance if anyone knows how to do this.

**Comment by L3MON4D3** [comment]
I think there was some mention of `opts` being disable if `init` is given in `layz.nvim` (which is used by kickstart IIRC), could you call `ls.setup({enable_autosnippets=true})` in `init` and see if that works?

**Comment by nchopin** [comment]
Hi,
Thanks a lot for getting back to me! I tried what you said, it did not work. However, I managed to fix the issue by doing something a bit weird. A few lines below in the `init.lua` of `kickstart.nvim`, I found this in the config of a related plugin:

```
      'saadparwaiz1/cmp_luasnip',

      -- Adds other completion capabilities.
      --  nvim-cmp does not ship with all sources by default. They are split
      --  into multiple repos for maintenance purposes.
      'hrsh7th/cmp-nvim-lsp',
      'hrsh7th/cmp-path',
    },
    config = function()
      -- See `:help cmp`
      local cmp = require 'cmp'
      local luasnip = require 'luasnip'
      luasnip.config.setup {}
```
See [here](https://github.com/nvim-lua/kickstart.nvim/blob/4120893b8a1f31a0957f2f891f7fbef73ddfb9b1/init.lua#L739C1-L751C30).

And I changed the last line to 
`      luasnip.config.setup({enable_autosnippets=true})`

It's not elegant, but it works. 

Should I raise an issue on the `kickstart.nvim` repo ? It seems strange to me that you need to mess with the configuration of another plugin to configure luasnip. But I am really a lua (and lazy) newbie, so there may be a better way I'm missing. 

Anyway, thanks again for taking the time to look into this (and for developing this great plugin).

**Reply by L3MON4D3** [reply]
Ahhh I see, yeah that's valid :D
I'd agree that putting the call to `ls.config.setup()` into a different plugins definition can be confusing, and this discussion alone proves it :)
Adressing this would certainly be an improvement, so it's a good idea to open an issue :+1:
